corehq/apps/domain/models.py

```diff
@@ -879,6 +889,47 @@ def get_daily_outbound_sms_limit(self):
         return 50000
 
 
+class SecureURLField(models.CharField):
+    def validate(self, value, model_instance):
+        super().validate(value, model_instance)
+
+        if not value.startswith('https'):
+            raise ValidationError(f'{value} is not a secure URL.')
```

Comment:
Will this block setting the value to an empty string? Or are validators not executed on empty values?

Also, this seems like it's enforcing a policy decision that may not match real world circumstances. If someone moved their domain to self-hosted and decided not to put an SSL certificate on it, are we not going to redirect to that domain? I think this should be a strong suggestion (warning if violated, probably implemented in UI JS) rather than a hard requirement.